Source/AssetRipper.Processing/UtilProcessors/PropertyNameProcessor.cs

```diff
@@ -0,0 +1,110 @@
+﻿using AssetRipper.Assets;
+using AssetRipper.Assets.Collections;
+using AssetRipper.Assets.Utils;
+using AssetRipper.Import.Structure.Assembly;
+using AssetRipper.Import.Structure.Assembly.Serializable;
+using AssetRipper.Import.Structure.GameStructure;
+using AssetRipper.SourceGenerated.Classes.ClassID_115;
+using ICSharpCode.SharpZipLib.Checksum;
+using System.Text;
+using static AssetRipper.Import.Structure.Assembly.Serializable.SerializableType;
+
+namespace AssetRipper.Processing.Utils
+{
+	/// <summary>
+	/// Attempts to process property name uint values
+	/// to their intended string value.
+	/// <para>Replicates Unity checksum usage for property names.</para>
+	/// </summary>
+	public class PropertyNameProcessor
+	{
+		public PropertyNameProcessor(GameStructure gameStructure)
+		{
+			GameStructure = gameStructure;
+		}
+
+		private readonly HashSet<long> pathIdsProcessed = new();
+		private readonly Dictionary<string, uint> cachedPropertyNames = new();
+
+		public GameStructure GameStructure { get; }
+
+		public IReadOnlyDictionary<string, uint> CachedPropertyNames => cachedPropertyNames;
+
+		public uint Process(string propertyName)
+		{
+			if (cachedPropertyNames.TryGetValue(propertyName, out uint value))
+			{
+				return value;
+			}
+
+			uint output = CrcUtils.CalculateDigestUTF8(propertyName);
+
+			cachedPropertyNames.Add(propertyName, output);
+
+			return output;
+		}
+
+		public void Process(IMonoScript script)
+		{
+			if (pathIdsProcessed.Contains(script.PathID))
+			{
+				return;
+			}
+
+			SerializableType? behaviour = script.GetBehaviourType(GameStructure.AssemblyManager);
+
+			if (behaviour is null)
+			{
+				return;
+			}
+
+			foreach (Field field in behaviour.Fields)
+			{
+				Process(field.Name);
+			}
```

Comment:
Can nonserializable fields ever be referenced by the GenericBinding? If so, `SerializableType` excludes those fields.